enclave_build/src/lib.rs

```diff
@@ -12,14 +12,71 @@ mod yaml_generator;
 use aws_nitro_enclaves_image_format::defs::{EifBuildInfo, EifIdentityInfo, EIF_HDR_ARCH_ARM64};
 use aws_nitro_enclaves_image_format::utils::identity::parse_custom_metadata;
 use aws_nitro_enclaves_image_format::utils::{EifBuilder, SignEnclaveInfo};
-use docker::DockerUtil;
+use docker::{DockerError, DockerUtil};
 use serde_json::json;
 use sha2::Digest;
 use std::collections::BTreeMap;
-use yaml_generator::YamlGenerator;
+use thiserror::Error;
+use yaml_generator::{YamlGenerator, YamlGeneratorError};
 
 pub const DEFAULT_TAG: &str = "1.0";
 
+#[derive(Debug, Error)]
+pub enum EnclaveBuildError {
+    #[error("Docker error: `{0:?}`")]
+    DockerError(DockerError),
+    #[error("Invalid path: `{0}`")]
+    PathError(String),
+    #[error("Image pull error: `{0}`")]
+    ImagePullError(String),
+    #[error("Image inspect failed")]
+    ImageInspectError,
+    #[error("Linuxkit error: `{0}`")]
+    LinuxKitError(String),
+    #[error("File operation error: `{0:?}`")]
+    FileError(std::io::Error),
+    #[error("Ramfs error: `{0:?}`")]
+    RamfsError(YamlGeneratorError),
+    #[error("Signature error: `{0}`")]
+    SignError(String),
+    #[error("Metadata error: `{0}`")]
+    MetadataError(String),
+    #[error("Unsupported architecture")]
+    UnsupportedArchError,
+    #[error("Getting image detail failed: `{0}`")]
+    ImageDetailError(String),
```

Comment:
I wanted to leave this comment in the other PR as well but forgot about it. Why do we use `ImageDetail` instead of `ImageMetadata` in the structure name/error variant etc. `detail` makes me think of something that is not important, and I think I would prefer the latter option.

Reply:
The structure name introduced in the next PR is `ImageDetails` to avoid confusion with the EIF metadata section. They are both metadata of some image type.

Reply:
Hmm okay, maybe we can find a better name though, but we can leave that for when we add the ImageDetails structure.